config/main.py

```diff
@@ -2969,6 +2969,56 @@ def reset(ctx, interface_name):
     cmd = "sudo sfputil reset {}".format(interface_name)
     clicommon.run_command(cmd)
 
+#
+# 'mpls' subgroup ('config interface mpls ...')
+#
+
+@interface.group()
+@click.pass_context
+def mpls(ctx):
+    """Add or remove MPLS"""
+    pass
+
+#
+# 'add' subcommand
+#
+
+@mpls.command()
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.pass_context
+def add(ctx, interface_name):
+    """Add MPLS towards the interface"""
+    config_db = ctx.obj["config_db"]
+    if clicommon.get_interface_naming_mode() == "alias":
+        interface_name = interface_alias_to_name(config_db, interface_name)
+        if interface_name is None:
+            ctx.fail("'interface_name' is None!")
+
+    table_name = get_interface_table_name(interface_name)
+    if table_name == "":
+        ctx.fail("'interface_name' is not valid. Valid names [Ethernet/PortChannel/Vlan/Loopback]")
+    config_db.set_entry(table_name, interface_name, {"mpls": "enable"})
+
+#
+# 'del' subcommand
+#
+
+@mpls.command()
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.pass_context
+def remove(ctx, interface_name):
+    """Remove MPLS from the interface"""
+    config_db = ctx.obj["config_db"]
+    if clicommon.get_interface_naming_mode() == "alias":
+        interface_name = interface_alias_to_name(config_db, interface_name)
+        if interface_name is None:
+            ctx.fail("'interface_name' is None!")
+
+    table_name = get_interface_table_name(interface_name)
+    if table_name == "":
+        ctx.fail("'interface_name' is not valid. Valid names [Ethernet/PortChannel/Vlan/Loopback]")
+    config_db.set_entry(table_name, interface_name, {"mpls": "disable"})
```

Comment:
i do not use mpls unit test.

Reply:
@lguohan I have examined unit tests for other similar CLI configuration commands.  The unit tests appear to use the command to set configuration, then verify the setting using output from CLI "show" commands.  MPLS enable/disable is not included in the output of any CLI "show" command, so there is not a similar mechanism for unit test.  Please suggest appropriate unit test.

Reply:
How is user expected to find out if "MPLS" is enabled on an interface? May 'show' command is required if not already there. 
```
show interface mpls <interface_name>
show interface mpls         <--- show all interfaces with MPLS enabled/disabled flag
```
At this point SONiC doesn't have a CLI to print all the properties of L3 interface.

Reply:
@smaheshm I have added "show interface mpls <interface_name>" to SONiC CLI.

Reply:
Thanks for adding the "show" command. Can you use it to write the unit test the MPLS config and show commands.